protocol-test-util/src/main/java/io/zeebe/test/broker/protocol/clientapi/ClientApiRule.java

```diff
@@ -40,31 +42,41 @@
 
 public class ClientApiRule extends ExternalResource {
 
-  private static final String DEFAULT_WORKER = "defaultWorker";
   public static final long DEFAULT_LOCK_DURATION = 10000L;
-
-  protected ClientTransport transport;
+  private static final String DEFAULT_WORKER = "defaultWorker";
+  private static final String DEFAULT_ADDRESS = "0.0.0.0";
+  private static final int DEFAULT_ATOMIX_PORT = 25617;
+  private static final String DEFAULT_CLUSTER_NAME = "zeebe-cluster";
 
   protected final int nodeId;
   protected final Supplier<SocketAddress> brokerAddressSupplier;
-
-  protected MsgPackHelper msgPackHelper;
-
   private final Int2ObjectHashMap<PartitionTestClient> testPartitionClients =
       new Int2ObjectHashMap<>();
   private final ControlledActorClock controlledActorClock = new ControlledActorClock();
-  private ActorScheduler scheduler;
-
+  private final Atomix atomix;
+  protected ClientTransport transport;
+  protected MsgPackHelper msgPackHelper;
   protected int defaultPartitionId = -1;
+  private ActorScheduler scheduler;
 
   public ClientApiRule(final Supplier<SocketAddress> brokerAddressSupplier) {
-    this.nodeId = 0;
-    this.brokerAddressSupplier = brokerAddressSupplier;
+    this(0, brokerAddressSupplier);
   }
 
   public ClientApiRule(final int nodeId, final Supplier<SocketAddress> brokerAddressSupplier) {
     this.nodeId = nodeId;
     this.brokerAddressSupplier = brokerAddressSupplier;
+    this.atomix =
+        Atomix.builder()
+            .withMemberId("clientApiRule")
+            .withClusterId(DEFAULT_CLUSTER_NAME)
+            .withProfiles(Profile.client())
+            .withMembershipProvider(
+                BootstrapDiscoveryProvider.builder()
+                    .withNodes(Address.from(DEFAULT_ADDRESS, DEFAULT_ATOMIX_PORT))
```

Comment:
This should be taken from the broker address supplier instead of the a static configuration. Therefore atomix probably can only be setup in the before method instead of the constructor.